Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -251,9 +221,47 @@ extension RawSyntax {
     return trailingTrivia?.sourceLength ?? .zero
   }
 
-  /// The length of this node including all of its trivia
-  var totalLength: SourceLength {
-    return leadingTriviaLength + contentLength + trailingTriviaLength
+  /// The length of this node excluding its leading and trailing trivia.
+  var contentLength: SourceLength {
+    return totalLength - leadingTriviaLength - trailingTriviaLength
+  }
+
+  /// Convenience function to create a RawSyntax when its byte length is not
+  /// known in advance, e.g. it is programmatically constructed instead of
+  /// created by the parser.
+  ///
+  /// This is a separate function than in the initializer to make it more
+  /// explicit and visible in the code for the instances where we don't have
+  /// the length of the raw node already available.
+  static func createAndCalcLength(kind: SyntaxKind, layout: [RawSyntax?],
+      presence: SourcePresence) -> RawSyntax {
+    let length: SourceLength
+    if case .missing = presence {
+      length = SourceLength.zero
+    } else {
+      length = layout.compactMap({ $0?.totalLength }).reduce(.zero, +)
```

Comment:
I wonder if single `reduce()` is faster than `compactMap().reduce()` chain.
```swift
length = layout.reduce(into: .zero) { (length, node) in
   node.map { length += $0.totalLength }
}
```

Reply:
I changed this to a for-loop, it's more readable to me.